echo/echo-rest/src/main/java/com/netflix/spinnaker/echo/config/RestConfig.java

```diff
@@ -83,6 +83,13 @@ RestUrls restServices(
 
     RestUrls restUrls = new RestUrls();
 
+    // Add null check to prevent NullPointerException when no endpoints are configured
+    if (restProperties.getEndpoints() == null) {
```

Comment:
I'm a little torn here.  Things are arguably better without this check, right?  Echo fails to start because echo is misconfigured.  As in, someone set `rest.enabled` to true, but didn't configure any endpoints.  Isn't failing fast the more helpful thing here?  That way folks have a (much) better chance of fixing their config, as opposed to running for a potentially long time before realizing they're not getting events.

Reply:
This is a second layer of security, but the   @Valid List<RestEndpointConfiguration> endpoints = new ArrayList<>(); should fix it! 
We can remove the if, if needed :)

Reply:
We disagree.  To me, this PR makes it easier to miss configuration errors.

Reply:
I kinda agree with DB - this SHOULD have been failing.  I THINK replacing the exception with a GOOD exception (InvalidConfig or similar) and a GOOD ERROR message ... and then just release notes "IF You've previously had rest.enable true without endpoints, this will not fail on this.".  

I can see SOME people making a mistake or an insert mistake... OR "we leave it default but only set the endpoint in prod" is the opposite argument.  That "we left it on everywhere but only set it in prod" is the only case I can think of... and if you're doing that, probably should just set rest.enabled in prod to true IMHO.  

Either way it DOES break things (apparently our internal systems left this on... and never set endpoints, which is where this came from).  AKA mistake on armory internals.

Reply:
I understand both of you, and I will close this PR! 
Maybe we can just add a comment for this in the next release, so people are aware of it! :) 

Thank you for taking the time to review and comment!